packages/ui/src/components/text.tsx

```diff
@@ -1,138 +1,184 @@
 import { Slot } from '@radix-ui/react-slot'
 import { cn } from '@utils/cn'
-import { cva } from 'class-variance-authority'
+import { cva, VariantProps } from 'class-variance-authority'
 
-const textVariants = cva('text-base', {
+type TextElement =
+  | 'p'
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'h6'
+  | 'span'
+  | 'label'
+  | 'legend'
+  | 'caption'
+  | 'figcaption'
+  | 'blockquote'
+  | 'pre'
+  | 'kbd'
+  | 'var'
+  | 'code'
+  | 'samp'
+  | 'cite'
+  | 'time'
+  | 'address'
+  | 'strong'
+  | 'abbr'
+  | 'em'
+
+const textVariants = cva('', {
   variants: {
+    variant: {
+      'heading-hero': 'font-heading-hero',
+      'heading-section': 'font-heading-section',
+      'heading-subsection': 'font-heading-subsection',
+      'heading-base': 'font-heading-base',
+      'heading-small': 'font-heading-small',
+      'body-normal': 'font-body-normal',
+      'body-single-line-normal': 'font-body-single-line-normal',
+      'body-strong': 'font-body-strong',
+      'body-single-line-strong': 'font-body-single-line-strong',
+      'body-code': 'font-body-code',
+      'caption-normal': 'font-caption-normal',
+      'caption-soft': 'font-caption-soft',
+      'caption-single-line-normal': 'font-caption-single-line-normal',
+      'caption-single-line-soft': 'font-caption-single-line-soft'
+    },
     align: {
       left: 'text-left',
       center: 'text-center',
       right: 'text-right'
     },
-    size: {
-      0: 'text-1',
-      1: 'text-2',
-      2: 'text-sm',
-      3: 'text-base',
-      4: 'text-xl',
-      5: 'text-2xl',
-      6: 'text-3xl',
-      7: 'text-4xl',
-      8: 'text-5xl',
-      9: 'text-6xl',
-      10: 'text-7xl',
-      11: 'text-8xl',
-      12: 'text-9xl',
-      13: 'text-lg'
-    },
-    trim: {
-      normal: '',
-      start: '',
-      end: '',
-      both: ''
-    },
     truncate: {
       true: 'truncate'
     },
-    weight: {
-      normal: 'font-normal',
-      medium: 'font-medium',
-      semibold: 'font-semibold',
-      bold: 'font-bold'
-    },
     color: {
-      primary: 'text-cn-foreground-1',
-      secondary: 'text-cn-foreground-2',
-      tertiary: 'text-cn-foreground-3',
-      tertiaryBackground: 'text-cn-foreground-3',
       inherit: 'text-inherit',
-      'foreground-4': 'text-cn-foreground-2',
-      'foreground-5': 'text-cn-foreground-3'
+      'foreground-1': 'text-cn-foreground-1',
+      'foreground-2': 'text-cn-foreground-2',
+      'foreground-3': 'text-cn-foreground-3',
+      disabled: 'text-cn-foreground-disabled',
+      success: 'text-cn-foreground-success',
+      warning: 'text-cn-foreground-warning',
+      danger: 'text-cn-foreground-danger'
     },
     wrap: {
       wrap: 'text-wrap',
       nowrap: 'text-nowrap',
       pretty: 'text-pretty',
       balance: 'text-balance'
     }
+  },
+  defaultVariants: {
+    variant: 'body-normal',
+    align: 'left',
+    color: 'foreground-2',
+    truncate: false
```

Comment:
should we consider truncating by default? in what situation would we want to avoid truncating if the text doesn't fit in the provided bounding box?

Reply:
@abhinavrastogi-harness truncate adds `white-space: nowrap;`, which breaks the normal flow of the text. Truncate should be added only in cases when the text should take no more than 1 line. Things like truncating or clamping solve specific problems, so they should be used only when needed